src/Core/Components/CounterBadge/FluentCounterBadge.razor.cs

```diff
@@ -83,27 +96,43 @@ public partial class FluentCounterBadge : FluentComponentBase, IDisposable
     [Parameter]
     public Color? Color { get; set; }
 
-    ///// <summary>
-    /////  If just a dot should be displayed without the count.
-    /////  Defaults to false.
-    ///// </summary>
-    //[Parameter]
-    //public bool Dot { get; set; } = false;
+    /// <summary>
+    ///  Gets or sets if just a dot should be displayed. Count will be ignored if this is set to true.
+    ///  Defaults to false.
+    /// </summary>
+    [Parameter]
+    public bool Dot { get; set; } = false;
 
     /// <summary>
     /// If the counter badge should be displayed when the count is zero.
     /// Defaults to false.
+    /// [Obsolete] This parameter will be removed in a future version. Use ShowWhen with a lambda expression instead
+    /// For getting the same behavior as before, you can use ShowWhen="@(Count => Count >= 0)"
     ///</summary>
+    [Obsolete("This parameter will be removed in a future version. Use ShowWhen with a lambda expression instead")]
     [Parameter]
     public bool ShowZero { get; set; } = false;
 
+    /// <summary>
+    /// Gets or sets if the counter badge is displayed based on the specified lambda expression.
+    /// For example to show the badge when the count greater than 4, use ShowWhen=@(Count => Count > 4)
+    /// Default the badge shows when the count is greater than 0.
+    /// </summary>
+    [Parameter]
+    public Func<int, bool> ShowWhen { get; set; } = Count => Count > 0;
+
     /// <summary>
     /// If an plus sign should be displayed when the <see cref="Count"/> is greater than <see cref="Max"/>.
     /// Defaults to true.
     ///</summary>
     [Parameter]
     public bool ShowOverflow { get; set; } = true;
 
+    public FluentCounterBadge()
```

Comment:
Why this empty constructor?
If required, could you move it before the properties?